redisenterprise/datadog_checks/redisenterprise/check.py

```diff
@@ -0,0 +1,320 @@
+            'big_del_ram',
+            'big_del_flash',
+        ]
+        stats = self._api_fetch_json(host, port, timeout, auth, verifyssl, "bdbs/stats/last", service_check_tags)
+        self.gauge('redisenterprise.database_count', len(stats), tags=service_check_tags, hostname=host)
+        for i in stats:
+            tgs = []
+            tgs.append('database:{}'.format(bdb_dict[int(i)]['name']))
+            # add the stats only available from the bdb_dict
+            self.gauge(
+                'redisenterprise.endpoints', bdb_dict[int(i)]['endpoints'], tags=tgs + service_check_tags, hostname=host
+            )
+            self.gauge(
+                'redisenterprise.memory_limit',
+                bdb_dict[int(i)]['limit'],
+                tags=tgs + service_check_tags,
+                hostname=host,
+            )
+            # derive our own stats from others
+            self.gauge(
+                'redisenterprise.used_memory_percent',
+                100 * stats[i]['used_memory'] / bdb_dict[int(i)]['limit'],
+                tags=tgs + service_check_tags,
+                hostname=host,
+            )
+            # derive our cache hit rate - be sure not to divide by 0
+            if (
+                stats[i]['read_hits'] + stats[i]['read_misses'] + stats[i]['write_hits'] + stats[i]['write_misses']
+            ) == 0:
+                self.gauge(
+                    'redisenterprise.cache_hit_rate',
+                    0.0,
+                    tags=tgs + service_check_tags,
+                    hostname=host,
+                )
+            else:
+                self.gauge(
+                    'redisenterprise.cache_hit_rate',
+                    100
+                    * (stats[i]['read_hits'] + stats[i]['write_hits'])
+                    / (
+                        stats[i]['read_hits']
+                        + stats[i]['read_misses']
+                        + stats[i]['write_hits']
+                        + stats[i]['write_misses']
+                    ),
+                    tags=tgs + service_check_tags,
+                    hostname=host,
+                )
+            # derive flash object percentage being sure that the key esists and is not 0
+            if 'bigstore_objs_flash' in stats[i].keys():
+                if stats[i]['bigstore_objs_flash'] > 0:
+                    self.gauge(
+                        'redisenterprise.bigstore_objs_percent',
+                        100
+                        * stats[i]['bigstore_objs_ram']
+                        / (stats[i]['bigstore_objs_ram'] + stats[i]['bigstore_objs_flash']),
+                        tags=tgs + service_check_tags,
+                        hostname=host,
+                    )
+
+            for j in stats[i].keys():
+                if j in gauges:
+                    self.gauge(
+                        'redisenterprise.{}'.format(j), stats[i][j], tags=tgs + service_check_tags, hostname=host
```

Comment:
```suggestion
                        'redisenterprise.{}'.format(j), stats[i][j], tags=tgs + service_check_tags
```
hostname is meant to be the agent host and not the check host. This variable is autofilled by the agent and should not be used except in very specific occasions